lib/ui/window.dart

```diff
@@ -457,7 +457,18 @@ class Window {
   ///
   ///  * [WidgetsBindingObserver], for a mechanism at the widgets layer to
   ///    observe when this value changes.
-  List<Locale> get locales => _locales;
+  List<Locale> get locales {
+    // Provide a minimal list of default locales for environments that do not
+    // support passing platform locales to Flutter. This will allow tests on
+    // these platforms to have similar expected behavior as Android and iOS where
+    // Locale passing is supported.
+    if ((_locales == null || _locales.isEmpty) && (Platform.isLinux || Platform.isMacOS || Platform.isFuchsia)) {
```

Comment:
Instead of a hard-coded list of platforms, I recommend using the FLUTTER_TEST environment variable (and only doing it in debug builds). See `foundation/platform.dart` in the framework repo.